src/views/clusteroverview/SettingsTab/ClusterTab/components/ResourceManagementSection/components/KernelSamepageMerging/KernelSamepageMerging.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { FC, useEffect, useState } from 'react';
+
+import HyperConvergedModel from '@kubevirt-ui/kubevirt-api/console/models/HyperConvergedModel';
+import HelpTextIcon from '@kubevirt-utils/components/HelpTextIcon/HelpTextIcon';
+import useHyperConvergeConfiguration, {
+  HyperConverged,
+} from '@kubevirt-utils/hooks/useHyperConvergeConfiguration';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import { k8sPatch } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Alert,
+  AlertVariant,
+  PopoverPosition,
+  Split,
+  SplitItem,
+  Switch,
+} from '@patternfly/react-core';
+
+import './KernelSamepageMerging.scss';
+
+const KernelSamepageMerging: FC = () => {
+  const { t } = useKubevirtTranslation();
+  const [isEnabled, setIsEnabled] = useState(false);
+  const [hyperConverge, hyperLoaded, hyperLoadingError] = useHyperConvergeConfiguration();
+  const [error, setError] = useState<Error>(null);
+
+  useEffect(() => {
+    if (!hyperLoaded || hyperLoadingError) {
+      setError(hyperLoadingError);
+    } else {
+      const ksmConfiguration = hyperConverge?.spec?.configuration?.ksmConfiguration;
+
+      setError(null);
+      setIsEnabled(!!(ksmConfiguration && isEmpty(ksmConfiguration?.nodeLabelSelector))); // Empty nodeLabelSelector will enable KSM on every node.
+    }
+  }, [hyperConverge, hyperLoaded, hyperLoadingError]);
+
+  const onKSMchange = async (value: boolean) => {
+    setIsEnabled(value);
+
+    k8sPatch<HyperConverged>({
+      data: [
+        {
+          op: 'replace',
+          path: `/spec/configuration/ksmConfiguration/nodeLabelSelector`,
+          value: value ? {} : null,
+        },
+      ],
+      model: HyperConvergedModel,
+      resource: hyperConverge,
+    })
+      .then(() => setIsEnabled(value))
+      .catch((err: Error) => setError(err));
```

Comment:
the function this is running is async , u can use await and try and catch

Reply:
I can but I am not sure. I've found 51 results in 44 files of our repo where we deal with async functions like this. And with this it is IMHO more obvious what and when happens.

Reply:
i dont know about other places in code, but this is wrong, this is anti pattern, u should remove the async or add await, no benefit at all in this way. don't just look on other places on the code and copy, or use as conformations.

Reply:
i suggest looking a bit about promises and seeing the difference before u decide this is a good way to handle this function. as there is no sense in this combination that you claim that is more clear.

Reply:
Ah, I see. I think we misunderstood each other again about what is the problem, but let me take a closer look at this again, and I'll solve it for sure..